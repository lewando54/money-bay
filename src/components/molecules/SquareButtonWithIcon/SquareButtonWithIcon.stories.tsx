/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { Meta,StoryObj } from '@storybook/react-native'

import SquareButtonWithIcon from './SquareButtonWithIcon'

const meta = {
    component: SquareButtonWithIcon,
    argTypes: {
        text: { control: 'text' },
        onclick: { action: 'clicked' },
        buttonBackgroundColor: { control: 'color' },
        circleBackgroundColor: { control: 'object' },
    }
} satisfies Meta<typeof SquareButtonWithIcon>

export default meta
type Story = StoryObj<typeof SquareButtonWithIcon>

export const Default: Story = {
    args: {
        text: 'Add new card',
        buttonBackgroundColor: '#fff',
        circleBackgroundColor: '#3EB290'
    }
}

export const Gradient: Story = {
    args: {
        text: 'Add new card',
        buttonBackgroundColor: '#fff',
        circleBackgroundColor: ['#3EB290', '#ff0']
    }
}
