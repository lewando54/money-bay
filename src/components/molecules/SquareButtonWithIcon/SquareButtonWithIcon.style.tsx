import { StyleSheet } from 'react-native'

import globalThemeStyle, { globalColors } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    outerContainer: {
        width: 150,
        aspectRatio: 15/13
    },
    innerContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        gap: 10
    },
    text: {
        ...globalThemeStyle.text_SemiBold,
        color: globalColors.MAIN_DARK,
        fontSize: 16
    }
})
