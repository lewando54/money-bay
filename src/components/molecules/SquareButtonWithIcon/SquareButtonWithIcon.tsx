import React from 'react'
import {  TouchableOpacity, View } from 'react-native'
import { AntDesign } from '@expo/vector-icons'

import { CustomText } from '@atoms'
import { CircleWithIcon, UniversalContainer } from '@atoms'

import squareButtonWithIconStyle from './SquareButtonWithIcon.style'

interface ISquareButtonWithIconProps {
    buttonBackgroundColor?: string
    circleBackgroundColor?: string
    icon?: React.ReactNode
    text?: string
    onClick?: () => void
    testID?: string
}

export default function SquareButtonWithIcon({
    buttonBackgroundColor='#fff',
    circleBackgroundColor='#3EB290',
    icon=<AntDesign size={20} name={'creditcard'} color={'#fff'}/>,
    text='Add new card',
    onClick,
    testID='square-button-with-icon'
}: ISquareButtonWithIconProps){
    return (
        <TouchableOpacity onPress={onClick} testID={testID} style={squareButtonWithIconStyle.outerContainer}>
            <UniversalContainer padding={20} height={'100%'} castShadow={true} backgroundColor={buttonBackgroundColor}>
                <View style={squareButtonWithIconStyle.innerContainer}>
                    <CircleWithIcon icon={icon} backgroundColor={circleBackgroundColor} size={50} />
                    <CustomText style={squareButtonWithIconStyle.text}>{text}</CustomText>
                </View>
            </UniversalContainer>
        </TouchableOpacity>
    )
}
