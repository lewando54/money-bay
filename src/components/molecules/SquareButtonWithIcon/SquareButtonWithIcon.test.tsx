import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react-native'
import SquareButtonWithIcon from './SquareButtonWithIcon'
describe('SquareButtonWithIcon', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<SquareButtonWithIcon />)
    expect(tree).toMatchSnapshot()
  })

  it('should call onClick when pressed', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(<SquareButtonWithIcon onClick={onClick} testID='test'/>)
    fireEvent.press(getByTestId('test'))
    expect(onClick).toBeCalled()
  })
})
