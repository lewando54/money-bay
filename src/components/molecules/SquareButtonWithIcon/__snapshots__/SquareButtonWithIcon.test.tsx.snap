// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareButtonWithIcon should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "aspectRatio": 1.1538461538461537,
      "opacity": 1,
      "width": 150,
    }
  }
  testID="square-button-with-icon"
>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "100%",
          "padding": 20,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#161966",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
      ]
    }
    testID="universal-container"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "gap": 10,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#3EB290",
              "borderRadius": 100,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            },
            false,
          ]
        }
      >
        <Text />
      </View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 16,
            },
          ]
        }
      >
        Add new card
      </Text>
    </View>
  </View>
</View>
`;
